time_manager/get_cal_time.F90

```diff
@@ -300,47 +311,63 @@ function get_cal_time(time_increment, units, calendar, permit_calendar_conversio
   base_time             = set_date(year+floor(time_increment)  ,month,day,hour,minute,second)
   base_time_plus_one_yr = set_date(year+floor(time_increment)+1,month,day,hour,minute,second)
   call get_time(base_time_plus_one_yr - base_time, second, day)
-  dt = (day*86400+second)*(time_increment-floor(time_increment))
-  increment_days = floor(dt/86400)
-  increment_seconds = int(dt - increment_days*86400)
+  dt = real(day*spd_int+second, r8_kind)*(time_increment-real(floor(time_increment), r8_kind))
+  increment_days = floor(dt/spd_real)
```

Comment:
Probably should be consistent here with the two instances of increment_days which are both using spd_int - or should they all be spd_real since dt is a real?